src/codecs/webp/extended.rs

```diff
@@ -112,18 +113,45 @@ impl ExtendedImage {
             type Item = ImageResult<Frame>;
 
             fn next(&mut self) -> Option<Self::Item> {
-                if let ExtendedImageData::Animation { frames, anim_info } = &self.image.image {
-                    let frame = frames.get(self.index);
-                    match frame {
-                        Some(anim_image) => {
-                            self.index += 1;
-                            ExtendedImage::draw_subimage(
-                                &mut self.canvas,
-                                anim_image,
-                                anim_info.background_color,
-                            )
+                if let ExtendedImageData::Animation {
+                    frames,
+                    anim_info,
+                    first_frame,
+                } = &self.image.image
+                {
+                    // Loop until a valid frame is found to skip invalid frames
+                    loop {
+                        let frame = frames.get(self.index);
```

Comment:
If you use the `?` operator (i.e. `frames.get(self.index)?`) then there is no need for the match below